src/libsync/owncloudpropagator.cpp

```diff
@@ -413,6 +413,64 @@ void OwncloudPropagator::resetDelayedUploadTasks()
     _delayedTasks.clear();
 }
 
+void OwncloudPropagator::adjustDeletedFoldersWithNewChildren(SyncFileItemVector &items)
+{
+    /* 
+       process each item that is new and is a directory and make sure every parent in its tree has the instruction CSYNC_INSTRUCTION_NEW
+       instead of CSYNC_INSTRUCTION_REMOVE 
+    */
+    for (const auto &item : items) {
+        if (item->_instruction != CSYNC_INSTRUCTION_NEW || item->_direction != SyncFileItem::Up || !item->isDirectory() || item->_file == QStringLiteral("/")) {
+            continue;
+        }
+
+        // #1 get root folder name for the current item that we need to reupload
+        const auto folderPathSplit = item->_file.split(QLatin1Char('/'), Qt::SkipEmptyParts);
+        if (folderPathSplit.isEmpty()) {
+            continue;
+        }
+        const auto itemRootFolderName = folderPathSplit.first();
+        if (itemRootFolderName.isEmpty()) {
+            continue;
+        }
+        // #2 iterate backwords (for optimization) and find the root folder by name
```

Comment:
are you sure ?
I would have expected the opposite
that in the `items` container `a` would be before `a/b`

Reply:
OK but that is not necessarily what you achieve here
you are always using std::rbegin(items) as starting point for the iteration so you will most probably iterate as many items as if you start from std::begin(items)

Reply:
@mgallien True! And, good catch. I will need to fix that.

Reply:
@matthieugallien Fixed